### DISCUSSION TITLE
[Design/Medical] Cloning Rework Concept

### DISCUSSION THREAD

**Discussion by JimKil3** [ParadiseSS13/Paradise#20532] [Documents]
### Foreword
Cloning is, at the moment, not very engaging. As the average doctor, you drag a corpse over, pop it in the miracle machine, wait a few minutes, then (maybe) treat some clone damage in the immediately adjacent cryotube and give the patient mutadone. As a chemist or CMO, you might make some cryoxadone and then some synthetic flesh for biomass, and while the reagent isn't _easy_ to produce it certainly isn't _hard_, and it'll usually last the entire shift, leaving chemists to go speedrun the rest of chem and sit around for >an hour.
### Current Mechanics Breakdown
- Cloning fixes all internal damage and revives a person.
- It requires biomass, usually synthetic meat, to be put into the cloning pod.
- Its only drawbacks are almost entirely mitigated by science upgrades and commonly-available medicines and techniques.

**Pros:**
- Cloning provides a noob-friendly way to revive past the five minute defib timer.
- Cloning makes healing especially damaged victims, like russian roulette losers or legion'd miners, not a slog.

**Cons:**
- It's _way_ easier to revive especially damaged people through cloning than through any other method.
- Cloning has no impactful drawbacks, compared to strange reagent's gibbing and organ damage and defibrillation's timer.
- It's not engaging or skillful for medbay to clone people.
### Proposed Changes
I propose that cloning is reworked to actually clone people; rather than fully healing someone, I believe cloning should copy over their exact state into the new body, but with the _option_ to replace / heal certain organs or limbs in the cloning menu at a higher cost.

Here's an example: A security officer has been bombed and left for dead. Their body has come into medbay just too late to defibrillate, and they're missing a leg, have a couple of bone fractures, a ruptured lung, and enough brute damage to gib them with SR.
Steps for current cloning to fix the unfortunate officer:
- Drag them to cloning.
- Strip them and put them into the cloning scanner.
- Press the button and wait.

Steps for my proposed changes: 
- Drag them to cloning.
- Strip them and put them into the cloning scanner.
- Select their heart (because it's necrotic; this would be _required_ for all cloning past 5 minutes) and lungs for replacement, at the cost of fresh organs inserted into the scanner, and see that the brute damage will cost a significant chunk of biomass to fix, then select that as well.
- Either clone them as is and fix their bones surgically later, or..
- Select the bones to fix via cloning as well, at a cost (which will be explained later).
- Clone, then defibrillate, fix clone damage and potential mutations, and take them to robotics to replace their leg, which can't be cloned (there's nothing _to_ clone!)

I believe this would make cloning significantly more engaging for doctors, as well as make other options like SR more appealing. To address the problem of requiring organs, or having unreplaceable missing limbs, the cloner could produce individual organs or limbs given a healthy scan of a patient (e.x., a doctor scans themselves and clones just their leg, then uses that for the officer), or the cloner could accept prosthetics to attach automatically during the cloning process.

As for fixing internal bleeding and bones, and to keep chemists occupied throughout a shift, two new advanced chemicals could be added and loaded into the cloner to be consumed for repairing these damages. Here's my initial ideas:
**Collagen Slurry**
_A gross, white paste, which is supposedly a component of surgical bone gel. It causes moderate toxin damage and slows down when ingested, but has a very low chance to fix broken bones._
- 1u Milk
- 1u Potassium
- 4u Oxygen
- 3u Phenol
- 3u Formaldehyde

**Sanguine Reagent**
_A crimson, near-black and seemingly congealed goo, rumored to have been developed by the Synthetic Union for unknown reasons. It can act as blood for any type, including exotic species, but paralyzes and causes severe toxins damage when exceeding its low overdose threshold._
- 1u Omnizine
- 1u Iron
- 1u Saline-Glucose Solution

### Addendum 1 (20230306)
**Sanguine Reagent (Alternate)**
_A crimson, near-black goo that creeps up the walls of the container. Through an unknown, potentially arcane mechanism, it spreads throughout an IPC's chassis on ingestion, spreading damage evenly throughout every limb and slowly healing them. If an IPC overdoses on Sanguine Reagent, their vision will go red and the healing will slightly increase, but after the reagent is out of their system they will collapse and suffer critical damage to their internals._
- 1u Omnizine
- 1u Synthflesh
- 1u Liquid Solder

### Addendum 2 (20230310)
**Sanguine Reagent (Current Working Concept)**
_A crimson, near-black goo that creeps up the walls of the container. It spreads throughout the interior of an IPC's chassis on ingestion, patching dents and fixing cables but corroding the metal. For organics, it quickly restores blood, but causes significant toxins damage._
- 1u Omnizine
- 1u Synthflesh
- 1u Liquid Solder

### Addendum 3 (20230520)
It's been a while since I've edited this document! Sanguine Reagent and collagen slurry, now known as Osseous Reagent, are detailed in #21148. Here's a few extra things I've been considering:
- Consuming cadavers for extra biomass
  - The cloner could be set to consume the body of the person to be cloned to save on biomass, depending on how much damage is left on their body. For example, if someone died of space kuru, consuming the body would give quite a bit of biomass towards the cloning, but if someone's missing multiple limbs more biomass would be required from external sources to clone. The CMO, or anyone with an emag, could also set the cloner to consume bodies without cloning them, providing an easy way to get extra biomass (or dispose of bodies).
- Not requiring limbs, just biomass
  - Limbs could be cloned at a high biomass cost, instead of forcing people to get cybernetics or monkey limbs. This would be to encourage a more 'fire and forget' mentality, which is what I believe the cloner should be used for.

**Plans for Implementation**
- [x] Rename Strange Reagent to Lazarus Reagent (so that there aren't two SRs) #20614 
- [x] Implement Collagen Slurry and Sanguine Reagent #21148
- [x] Change the cloner itself

**Comment by CinnamonSnowball** [comment]
Personally I like pretty much everything about this. Looks like a great system to make cloning more engaging and require some more effort from Medbay to set up, without affecting the speed too much and remaining a good "fire-and-forget" revival method for huge emergencies.

Something that might be an interesting idea on top of all this:
What if the cloner only allowed you to use a certain amount of biomass per cloning procedure, as a sort of "NT-mandated efficiency-preserving safety mechanism"?

That way, if the patient is really messed up and has tons of IBs and fractures (dead miners, for example), as a doctor you're *forced* to deal with some of the injuries before/after cloning instead of only relying on the magic cloning pod, since the cloner simply won't let you use enough biomass to fix everything. It would also introduce some interesting decision-making along the lines of "do I fix the IBs via cloning and fix the ruptured lungs on my own since that's just one surgery, or do I leave the IBs for myself since they're quick to do in bulk", etc.

Conveniently, this shouldn't noticeably slow down cloning revival in emergency situations too much, since in such cases typically you already have a doctor overseeing the cloning lab - with this tweak, they could get a small OR setup going to fix up things that can be done via surgery while they wait for the current cloning process to finish (or even to simply save on biomass).
